matrix/plugins/glitch/main.py

```diff
@@ -0,0 +1,97 @@
+import asyncio
+import functools
+import logging
+import time
+import yaml
+
+from .selectors import Selectors
+from .actions import Actions
+from .plan import generate_plan, validate_plan
+
+
+log = logging.getLogger("glitch")
+DEFAULT_PLAN_NAME = "glitch_plan.{}.yaml"
+
+
+def default_resolver(model, kind, name):
+    if kind not in ["application", "unit", "model", "controller", "relation"]:
+        return None
+    entities = getattr(model, kind + "s")
+    obj = entities[name]
+    return obj
+
+
+# XXX: this most likely will need an async def
+# depending on libjuju
```

Comment:
I don't think so.  All this is doing is accessing attributes on the model.  The model is updated in the background by a watcher, so accessing the attributes doesn't make any calls to the API and is entirely synchronous.